### PR TITLE
feat(CodeBuild): Ensure source repository URLs do not contain sensitive credentials

### PR DESCRIPTION
**Pull request by MarioRgzLpz**
### Context

To improve security and adhere to best practices for code repository management, we are adding a new check within the CodeBuild framework. This check ensures that source repository URLs (right now only Bitbucket) do not contain sensitive credentials like authentication tokens or username and password.

### Description

I added `codebuild_bitbucket_urls_no_sensitive_credentials `check with respective unit tests. In order to that I modified `codebuild_service.py` to add two new attributes `source` `secondary_sources` to the `Project` model, a new model `Source` and added respective test to `codebuild_service_test.py`.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
As the title of the check in Security Hub is "CodeBuild Bitbucket source repository URLs should not contain sensitive credentials", I would make this check to analyze every repository URL and not only BitBucket URL. And also using the detect secrets library.

**Comment by MrCloudSec** [comment]
Please name the check `codebuild_project_source_repo_url_no_sensitive_credentials ` and create a BaseModel called Source to fill it with the information needed.

File: prowler/providers/aws/services/codebuild/codebuild_project_source_repo_url_no_sensitive_credentials/codebuild_project_source_repo_url_no_sensitive_credentials.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        findings = []
+        token_pattern = re.compile(r"https://x-token-auth:[^@]+@bitbucket\.org/.+\.git")
+        user_pass_pattern = re.compile(r"https://[^:]+:[^@]+@bitbucket\.org/.+\.git")
+        for project in codebuild_client.projects.values():
+            report = Check_Report_AWS(self.metadata())
+            report.region = project.region
+            report.resource_id = project.name
+            report.resource_arn = project.arn
+            report.status = "PASS"
+            report.status_extended = f"CodeBuild project {project.name} does not contain sensitive credentials in source repository URLs."
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/codebuild/codebuild_project_source_repo_url_no_sensitive_credentials/codebuild_project_source_repo_url_no_sensitive_credentials.py]:
```suggestion
            report.status_extended = f"CodeBuild project {project.name} does not contain sensitive credentials in any source repository URLs."
```

File: prowler/providers/aws/services/codebuild/codebuild_project_source_repo_url_no_sensitive_credentials/codebuild_project_source_repo_url_no_sensitive_credentials.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            secrets_found = []
+
+            if project.source and project.source.type == "BITBUCKET":
+                if token_pattern.match(project.source.location):
+                    secrets_found.append(f"Token in URL {project.source.location}")
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/codebuild/codebuild_project_source_repo_url_no_sensitive_credentials/codebuild_project_source_repo_url_no_sensitive_credentials.py]:
```suggestion
                    secrets_found.append(f"Token in {project.source.type} URL {project.source.location}")
```

File: prowler/providers/aws/services/codebuild/codebuild_project_source_repo_url_no_sensitive_credentials/codebuild_project_source_repo_url_no_sensitive_credentials.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                elif user_pass_pattern.match(project.source.location):
+                    secrets_found.append(
+                        f"Basic Auth Credentials in URL {project.source.location}"
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/codebuild/codebuild_project_source_repo_url_no_sensitive_credentials/codebuild_project_source_repo_url_no_sensitive_credentials.py]:
```suggestion
                        f"Basic Auth Credentials in {project.source.type} URL {project.source.location}"
```

File: prowler/providers/aws/services/codebuild/codebuild_project_source_repo_url_no_sensitive_credentials/codebuild_project_source_repo_url_no_sensitive_credentials.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+                    )
+            for url in project.secondary_sources:
+                if url.type == "BITBUCKET":
+                    if token_pattern.match(url.location):
+                        secrets_found.append(f"Token in URL {url.location}")
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/codebuild/codebuild_project_source_repo_url_no_sensitive_credentials/codebuild_project_source_repo_url_no_sensitive_credentials.py]:
```suggestion
                        secrets_found.append(f"Token in {url.type} URL {url.location}")
```

File: prowler/providers/aws/services/codebuild/codebuild_project_source_repo_url_no_sensitive_credentials/codebuild_project_source_repo_url_no_sensitive_credentials.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+                    elif user_pass_pattern.match(url.location):
+                        secrets_found.append(
+                            f"Basic Auth Credentials in URL {url.location}"
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/codebuild/codebuild_project_source_repo_url_no_sensitive_credentials/codebuild_project_source_repo_url_no_sensitive_credentials.py]:
```suggestion
                            f"Basic Auth Credentials in {url.type} URL {url.location}"
```

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
Please, review my changes for a more detailed status extended.

**Comment by MarioRgzLpz** [comment]
@sergargar All the changes requested done in commit  64bade9.

**Review by MrCloudSec — APPROVED** [review]
❤️

**Pull request merged**